src/System.Management.Automation/utils/Verbs.cs

```diff
@@ -1311,6 +1309,107 @@ static Verbs()
+
+        internal static IEnumerable<VerbInfo> FilterByVerbsAndGroups(
+            string[] verbs,
+            string[] groups)
+        {
+            if (groups is null)
+            {
+                foreach (VerbInfo verb in FilterVerbs(verbs))
+                {
+                    yield return verb;
+                }
+
+                yield break;
+            }
+
+            foreach (VerbInfo verb in FilterVerbsWithGroups(verbs, groups))
+            {
+                yield return verb;
+            }
+        }
+
+        private static IEnumerable<VerbInfo> FilterVerbsWithGroups(
+            string[] verbs,
+            string[] groups)
+        {
+            Collection<WildcardPattern> verbPattern = SessionStateUtilities.CreateWildcardsFromStrings(
+                verbs ?? new string[] { "*" },
+                WildcardOptions.IgnoreCase);
+
+            foreach (Type verbType in VerbTypes)
+            {
+                string verbGroupName = GetVerbGroupDisplayName(verbType);
+
+                bool hasVerbGroup = SessionStateUtilities.CollectionContainsValue(
+                    groups,
+                    verbGroupName,
+                    StringComparer.OrdinalIgnoreCase);
+
+                if (hasVerbGroup)
+                {
+                    foreach (VerbInfo verb in FilterVerbsByWildCardPattern(verbType, verbGroupName, verbPattern))
+                    {
+                        yield return verb;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<VerbInfo> FilterVerbs(string[] verbs)
+        {
+            Collection<WildcardPattern> verbPattern = SessionStateUtilities.CreateWildcardsFromStrings(
+                verbs ?? new string[] { "*" },
+                WildcardOptions.IgnoreCase);
+
+            foreach (Type verbType in VerbTypes)
+            {
+                string verbGroupName = GetVerbGroupDisplayName(verbType);
+
+                foreach (VerbInfo verb in FilterVerbsByWildCardPattern(verbType, verbGroupName, verbPattern))
+                {
+                    yield return verb;
+                }
+            }
+        }
+
+        private static IEnumerable<VerbInfo> FilterVerbsByWildCardPattern(
+            Type verbType,
+            string verbGroupName,
+            IEnumerable<WildcardPattern> patterns)
+        {
+            foreach (FieldInfo field in verbType.GetFields())
+            {
+                bool matchesWildCardPattern = SessionStateUtilities.MatchesAnyWildcardPattern(
+                    field.Name,
+                    patterns,
+                    defaultValue: false);
+
+                if (field.IsLiteral && matchesWildCardPattern)
+                {
+                    yield return new VerbInfo
+                    {
+                        Verb = field.Name,
+                        AliasPrefix = VerbAliasPrefixes.GetVerbAliasPrefix(field.Name),
+                        Group = verbGroupName,
```

Comment:
Why do we need `verbGroupName` parameter? I suggest to remove it.
```suggestion
                        Group = GetVerbGroupDisplayName(verbType),
```

Reply:
Fixed [aae8518](https://github.com/PowerShell/PowerShell/pull/20286/commits/aae85180494051730422c9fa6d0871f1b8803e5b)